### PR TITLE
Fix ravepath

### PR DESCRIPTION
**Pull request by pinkenburg**
Hello All,
just some background first, I am the computing coordinator of the sPHENIX experiment which is currently being build at RHIC at BNL. Our current tracking is using genfit, we are also using rave for vertex reconstruction. So we are definitely interested in seeing rave support staying in genfit (I know that rave itself is somewhat orphaned but we got it to compile and work for us).
The main problem we encountered building GenFit with rave is that setting RAVEPATH did not do the trick. The current CMakeLists.txt assumes a .pc file from Rave which does not seem part of that distribution anymore. Rave does create and install a RaveConfig.cmake instead which contains all information needed. So using find_package(Rave REQUIRED) rather than pkg_search_module(RAVE rave REQUIRED), then adding ${Rave_DEFINITIONS} to the CFLAGS and ${Rave_INCLUDE_DIRS} to the GF_INC_DIRS does the trick. Another small problem for rave was that further downstream -I${RAVE_INCLUDE_DIRS_I} was added to the linker flags which lead to a dangling -I in a link.txt.
This pull request works now when setting RAVEPATH to the rave installation. If RAVEPATH is not set it does not add support for rave to genfit.
Chris

### CODE REVIEW

**Comment by nils-braun** [comment]
Nice! Thanks for this - this is really useful.
Unfortunately (for the genfit project), we at Belle II have our own building structure for genfit2, so we barely test the CMake file with the newest versions... :-( 
I have tested the cmake file and it seems to find rave (and also compiling it).

However, I still get a linker error and many undefined references to rave::*. Will look into this today.

**Comment by nils-braun** [comment]
Ah, I have found the issue. At least in the current RAVE version, I can not find a RAVE_LDFLAGS variable. Where is yours defined?

**Comment by nils-braun** [comment]
Ok, how about the following changes:

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 76f4399..3ceb107 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,24 +50,15 @@ IF(DEFINED ENV{RAVEPATH})
        find_package(Rave REQUIRED)
        
        SET(RAVE True)
-ELSEIF(DEFINED RAVE_LDFLAGS)
-       IF(DEFINED RAVE_INCLUDE_DIRS)
-               IF(DEFINED RAVE_CFLAGS)
-                       MESSAGE("RAVE_LDFLAGS, RAVE_INCLUDE_DIRS and RAVE_CFLAGS are set. Try to build GFRave.") 
-                       SET(RAVE True)
-               ENDIF()
-       ENDIF()
 ELSE()
        MESSAGE("You did not define the environment variable RAVEPATH which is needed to find rave. GFRave will not be built.") 
 ENDIF()
 
 IF(DEFINED RAVE)
        # turn comma separated lists into space separated strings
-       string (REPLACE ";" " " Rave_LDFLAGS_STR "${Rave_LDFLAGS}")
-       string (REPLACE ";" " " Rave_INCLUDE_DIRS_STR "${Rave_INCLUDE_DIRS}")
-       string (REPLACE ";" " " Rave_CFLAGS_STR "${Rave_CFLAGS}")
-       SET (CMAKE_CXX_FLAGS ${Rave_DEFINITIONS} ${Rave_CFLAGS_STR})
-       
+       get_filename_component(Rave_LDFLAGS ${Rave_LIB}, DIRECTORY)
+       SET(Rave_LDFLAGS_STR "-L${Rave_LDFLAGS} -lRaveBase")
+       SET (CMAKE_CXX_FLAGS ${Rave_DEFINITIONS})       
        SET(GF_INC_DIRS
                ./GFRave/include/ ${Rave_INCLUDE_DIRS}
        )
@@ -299,7 +290,7 @@ if(DEFINED RAVE)
        TARGET_LINK_LIBRARIES(
                        ${PROJECT_NAME}
                        ${ROOT_LIBS}
-                       ${RAVE_LDFLAGS_STR}
+                       ${Rave_LDFLAGS_STR}
        )
 else()
        TARGET_LINK_LIBRARIES(${PROJECT_NAME} ${ROOT_LIBS})
```

first, I do remove the older version of the RAVE config getting, as it is not supported anymore with the new variables. Second, I use the variable Rave_LDFLAGS_STR, but I need to "construct it" by myself, by adding the path of the Rave Lib with -L and the name with -l.
Could be probably done better, I know...

**Comment by pinkenburg** [comment]
Thank you for looking into this and sorry - it's my first encounter with cmake and I overlooked that the LDFLAGS for the rave libs got lost.

Your proposed  change would work but I think I have something simpler. RaveConfig.cmake sets Rave_LIB to libRaveBase.so (with its full path). Just adding ${Rave_LIB} to the TARGET_LINK_LIBRARIES is sufficient to resolve the library dependencies on Rave, ldd libgenfit2.so shows now all Rave libs.

I also preserved the capability to build genfit with Rave by using command line flags if RAVEPATH is not set (in case RaveConfig.cmake is not available). There was a mixed use of RAVE vs Rave for those flags, they are now consistently called Rave_XXX which hopefully simplifies things.

Last not least, I changed the status of the Rave related messages from important to STATUS which makes them much less alarming (and was one major reason for the desire to remove Rave). Now they are just regular configuration related messages.

File: CMakeLists.txt [hunk 1]
```diff
@@ -300,8 +299,8 @@ if(DEFINED RAVE)
 	TARGET_LINK_LIBRARIES(
 			${PROJECT_NAME}
 			${ROOT_LIBS}
-			${RAVE_LDFLAGS_STR}
-			-I${RAVE_INCLUDE_DIRS_I}
+			${Rave_LIB}
+                        ${Rave_LDFLAGS_STR}
```
[the remark this answers is not in the document]
Comment (pinkenburg) [reply CMakeLists.txt]: sure. Old habit, I accidentally used spaces instead of tabs. fixed

**Pull request merged**